t/unit-tests/t-reftable-pq.c

```diff
@@ -59,9 +59,51 @@ static void test_pq(void)
 	merged_iter_pqueue_release(&pq);
 }
 
+
+static void test_pq_index(void)
+{
+	struct merged_iter_pqueue pq = { NULL };
+	struct reftable_record recs[14];
+	char *last = NULL;
+	int N = ARRAY_SIZE(recs), i;
+
+	for (i = 0; i < N; i++) {
+		reftable_record_init(&recs[i], BLOCK_TYPE_REF);
+		recs[i].u.ref.refname = xstrdup("refs/heads/master");
+	}
+
+	for (i = 0; i < N; i++) {
+		struct pq_entry e = {
+			.rec = &recs[i],
+			.index = i,
+		};
+
+		merged_iter_pqueue_add(&pq, &e);
+		merged_iter_pqueue_check(&pq);
+	}
+
+	i = N - 1;
+	while (!merged_iter_pqueue_is_empty(pq)) {
+		struct pq_entry e = merged_iter_pqueue_remove(&pq);
+		merged_iter_pqueue_check(&pq);
+
+		check(reftable_record_type(e.rec) == BLOCK_TYPE_REF);
+		check_int(e.index, ==, i);
```

Comment:
Question: why don't we add the comparison of the index to the existing test instead of adding an entirely new one? Doesn't feel like we're exercising anything that would be particularly different here, we only verify more of our assumptions.

Reply:
We need an entirely different set of test input for index-based comparison than what we have in the current test, so I thought it would be better to write a new test to preserve readability than to clutter the existing one with multiple inputs.